ci-operator/step-registry/baremetal/lab/agent/day2/baremetal-lab-agent-day2-commands.sh

```diff
@@ -0,0 +1,263 @@
+    exit 0
+fi
+
+if [ "${ADDITIONAL_WORKERS_DAY2}" != "true" ]; then
+    echo "Skipping as the additional nodes have been provisioned at installation time."
+    exit 0
+fi
+
+function mount_virtual_media() {
+  local host="${1}"
+  local iso_path="${2}"
+  echo "Mounting the ISO image in #${host} via virtual media..."
+  timeout -s 9 10m ssh "${SSHOPTS[@]}" "root@${AUX_HOST}" mount.vmedia "${host}" "${iso_path}"
+  local ret=$?
+  if [ $ret -ne 0 ]; then
+    echo "Failed to mount the ISO image in #${host} via virtual media."
+    touch /tmp/virtual_media_mount_failure
+    return 1
+  fi
+  return 0
+}
+
+function get_ready_nodes_count() {
+  oc get nodes \
+    -o jsonpath='{range .items[*]}{.metadata.name}{","}{.status.conditions[?(@.type=="Ready")].status}{"\n"}{end}' | \
+    grep -c -E ",True$"
+}
+
+function approve_csrs() {
+  while [[ ! -f '/tmp/scale-out-complete' ]]; do
+    sleep 30
+    echo "approve_csrs() running..."
+    oc get csr -o go-template='{{range .items}}{{if not .status}}{{.metadata.name}}{{"\n"}}{{end}}{{end}}' \
+      | xargs --no-run-if-empty oc adm certificate approve || true
+  done
+}
+
+function wait_for_nodes_readiness()
+{
+  local expected_nodes=${1}
+  local max_retries=${2:-10}
+  local period=${3:-5}
+  for i in $(seq 1 "${max_retries}") max; do
+    if [ "${i}" == "max" ]; then
+      echo "[ERROR] Timeout reached. ${expected_nodes} ready nodes expected, found ${ready_nodes}... Failing."
+      return 1
+    fi
+    sleep "${period}m"
+    monitor_workers &
+    ready_nodes=$(get_ready_nodes_count)
+    if [ x"${ready_nodes}" == x"${expected_nodes}" ]; then
+        echo "[INFO] Found ${ready_nodes}/${expected_nodes} ready nodes, continuing..."
+        return 0
+    fi
+    echo "[INFO] - ${expected_nodes} ready nodes expected, found ${ready_nodes}..." \
+      "Waiting ${period}min before retrying (timeout in $(( (max_retries - i) * (period) ))min)..."
+  done
+}
+
+function monitor_workers()
+{
+  local DAY2_IPS=""
+
+  # shellcheck disable=SC2154
+  for bmhost in $(yq e -o=j -I=0 '.[]' "${SHARED_DIR}/hosts.yaml"); do
+    # shellcheck disable=SC1090
+    . <(echo "$bmhost" | yq e 'to_entries | .[] | (.key + "=\"" + .value + "\"")')
+    if [[ "${name}" == *-a-* ]] && [ "${ADDITIONAL_WORKERS_DAY2}" == "true" ]; then
+      DAY2_IPS+="${ip},"
+    fi
+  done
+  DAY2_IPS=${DAY2_IPS%,}
+  /tmp/oc adm node-image monitor --ip-addresses "${DAY2_IPS}" -a "${day2_pull_secret}" --insecure=true
+}
+
+SSHOPTS=(-o 'ConnectTimeout=5'
+  -o 'StrictHostKeyChecking=no'
+  -o 'UserKnownHostsFile=/dev/null'
+  -o 'ServerAliveInterval=90'
+  -o LogLevel=ERROR
+  -i "${CLUSTER_PROFILE_DIR}/ssh-key")
+
+#BASE_DOMAIN=$(<"${CLUSTER_PROFILE_DIR}/base_domain")
+#PULL_SECRET_PATH=${CLUSTER_PROFILE_DIR}/pull-secret
+
+
+INSTALL_DIR="${INSTALL_DIR:-/tmp/installer}"
+mkdir -p "${INSTALL_DIR}"
+day2_pull_secret="${INSTALL_DIR}/day2_pull_secret"
+
+
+cat > "${INSTALL_DIR}/nodes-config.yaml" <<EOF
```

Comment:
This part is done in other install scripts. I wonder if we can incorporate the creation of this file in this scripts by writing two files when ADDITIONAL_WORKERS_DAY2 is not true.
The second file would be used here directly without the need to duplicate code.

You'd use SHARED_DIR instead of INSTALL_DIR, especially here as we are past the installation

Reply:
This file default name(nodes-config.yaml) and format is not quite the same as agent-config, I would like to stay the code in this script.

Reply:
We are trying to switch to use patches mostly instead of infinite branches here. If you continue this way, you may need a lot more IFs in the future to address all the possible environment scenarios.
What doesn't work for you in a change like this for example? https://github.com/aleskandro/openshift-release/commit/11c521013101dced5586fddb8f0b39fdf33957b4

Reply:
Ok, got it, I will update dynamic/static conf scripts.